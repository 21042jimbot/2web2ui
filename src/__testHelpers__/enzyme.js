export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
