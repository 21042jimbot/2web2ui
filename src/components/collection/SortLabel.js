import React from 'react';
import { ArrowDropDown, ArrowDropUp } from '@sparkpost/matchbox-icons';
import { UnstyledLink } from 'src/components/matchbox';
import cx from 'classnames';

import styles from './SortLabel.module.scss';

const SortLabel = ({ label, direction, ...rest }) => {
  const classes = cx(
    styles.SortLabel,
    direction === 'asc' && styles.asc,
    direction === 'desc' && styles.desc,
  );

  return (
    <UnstyledLink className={classes} {...rest}>
      <span>{label}</span>
      <span className={styles.Up}>
        <ArrowDropUp />
      </span>
      <span className={styles.Down}>
        <ArrowDropDown />
      </span>
    </UnstyledLink>
  );
};

export default SortLabel;
