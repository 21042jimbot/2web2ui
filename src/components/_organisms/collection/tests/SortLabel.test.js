import React from 'react';
import SortLabel from '../SortLabel';
import { shallow } from 'enzyme';

describe('Collection SortLabel Component', () => {
  const props = {
    direction: 'asc',
    label: 'Table Header Label',
    anotherprop: 'anotherProp'
  };

  // Can't test more than this
  // Need to find out how to get css-modules rendered in enzyme
  it('should render correctly', () => {
    const wrapper = shallow(<SortLabel {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
