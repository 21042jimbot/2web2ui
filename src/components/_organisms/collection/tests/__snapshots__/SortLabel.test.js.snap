// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection SortLabel Component should render correctly 1`] = `
<UnstyledLink
  anotherprop="anotherProp"
  className="SortLabel asc"
>
  <span>
    Table Header Label
  </span>
  <span
    className="Up"
  >
    <ArrowDropUp />
  </span>
  <span
    className="Down"
  >
    <ArrowDropDown />
  </span>
</UnstyledLink>
`;
